examples/apps/generic_client/generic_client.py

```diff
@@ -465,10 +416,14 @@ def Main(args=None):
                     res['result'], session_key, session_iv)
             logger.info("\nDecrypted response:\n {}"
                         .format(decrypted_res))
+    # Handle worker key refresh scenario
+    elif "error" in res and \
+            res["error"]["code"] == \
+            WorkOrderStatus.WORKER_ENCRYPT_KEY_REFRESHED:
+        logger.error("Worker Key refreshed. Retrieving latest Worker details")
+        _start_client(config, options, jrpc_req_id, worker_id, worker_registry)
```

Comment:
So `_start_client()` is being called recursively here ? If so, please modify the logs. The logs is only mentioning about retrieving latest worker details. `_start_client()` also resubmits the work order.

Reply:
Its not called recursively. Only if work order execution fails due to worker key refresh during the submission of work order, the error scenario is handle by retrieving latest worker details(after key refresh) and submit the work order again.